pkg/quarterdeck/api/v1/client.go

```diff
@@ -715,9 +732,16 @@ func (s *APIv1) Do(req *http.Request, data interface{}, checkStatus bool) (rep *
 
 	// Deserialize the JSON data from the body
 	if data != nil && rep.StatusCode >= 200 && rep.StatusCode < 300 && rep.StatusCode != http.StatusNoContent {
-		// Check the content type to ensure data deserialization is possible
-		if ct := rep.Header.Get("Content-Type"); ct != contentType {
-			return rep, fmt.Errorf("unexpected content type: %q", ct)
+		ct := rep.Header.Get("Content-Type")
+		if ct != "" {
+			mt, _, err := mime.ParseMediaType(ct)
+			if err != nil {
+				return nil, fmt.Errorf("malformed content-type header: %w", err)
+			}
+
+			if mt != accept {
+				return nil, fmt.Errorf("unexpected content type: %q", mt)
+			}
 		}
```

Comment:
This is a more robust content-type checking that I wish I knew about last year when we implemented this!